### DISCUSSION TITLE
Delay Messages in a File Reader

### DISCUSSION THREAD

**Discussion by JJWest12** [nextgenhealthcare/connect#5115] [Q&A]
I have a file reading channel that is obviously reading from a file located in the folders on the control panel. My channel is taking data from that file and loading it in to a database. The issue I am having however is that I keep getting "The server sent HTTP status code 404: Not Found" errors when trying to load info in to the db. I am think that this is because the channel is reading the file quicker than the db can take in the information. What is the best way of putting in a delay in between reading messages from the same file?

**Comment by pacmano1** [comment]
You mixed a lot of stuff there.

What is control panel in this context?
What is issuing a 404, which is a web response?
What does "load info into the db" in the context of the 404 mean?

Screenshots?  Some code?

**Comment by franklwambo** [comment]
You could break the data flow to two channels. The first channel to have a file reader source and a TCP sender destination (trace the TCP sender settings on the destination connector and check the ignore response checkbox ). The second channel can be a TCP listener source (listening to the first channel) and a database writer destination. Under the source settings of the source connector of the second channel trace the portion labelled  source queue then on the drop down select ON(respond before processing). In that way, the TCP listener can grab many messages then queue them (using some internal mechanism) for DB save. There might be other better ways though.

**Reply by pacmano1** [reply]
IMHO, the problem is not stated clearly enough to suggest a solution.
